tests/DoctrineORMModuleTest/Service/DBALConnectionFactoryTest.php

```diff
@@ -0,0 +1,120 @@
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * This software consists of voluntary contributions made by many individuals
+ * and is licensed under the MIT license. For more information, see
+ * <http://www.doctrine-project.org>.
+ */
+
+namespace DoctrineORMModuleTest\Service;
+
+use PHPUnit_Framework_TestCase;
+use DoctrineORMModule\Service\DBALConnectionFactory;
+use Doctrine\DBAL\Types\Type;
+use Doctrine\Common\Cache\ArrayCache;
+use Doctrine\Common\EventManager;
+use Zend\ServiceManager\ServiceManager;
+use DoctrineORMModule\Service\ConfigurationFactory;
+
+class DBALConnectionFactoryTest extends PHPUnit_Framework_TestCase
+{
+    /**
+     * @var ServiceManager
+     */
+    protected $serviceManager;
+    /**
+     * @var DBALConnectionFactory
+     */
+    protected $factory;
+
+    /**
+     * {@inheritDoc}
+     */
+    public function setUp()
+    {
+        $this->serviceManager = new ServiceManager();
+        $this->factory = new DBALConnectionFactory('orm_default');
+        $this->serviceManager->setService('doctrine.cache.array', new ArrayCache());
+        $this->serviceManager->setService('doctrine.eventmanager.orm_default', new EventManager());
+    }
+
+    public function testDoctrineMappingTypeReturnCorrectParent()
+    {
+        $config = array(
+            'doctrine' => array(
+                'connection' => array(
+                    'orm_default' => array(
+                        'driverClass'   => 'Doctrine\DBAL\Driver\PDOSqlite\Driver',
+                        'params' => array(
+                            'memory' => true,
+                        ),
+                        'doctrineTypeMappings' => array(
+                            'money' => "string"
+                        ),
+                    )
+                ),
+            ),
+        );
+        $configurationMock = $this->getMockBuilder("Doctrine\ORM\Configuration")
+            ->disableOriginalConstructor()
+            ->getMock();
+        $this->serviceManager->setService('doctrine.configuration.orm_default', $configurationMock);
+        $this->serviceManager->setService('Config', $config);
+        $this->serviceManager->setService('Configuration', $config);
+        $dbal = $this->factory->createService($this->serviceManager);
+        $this->assertSame("string", $dbal->getDatabasePlatform()->getDoctrineTypeMapping("money"));
+    }
+
+    public function testDoctrineAddCustomCommentedType()
+    {
+        $config = array(
+            'doctrine' => array(
+                'connection' => array(
+                    'orm_default' => array(
+                        'driverClass'   => 'Doctrine\DBAL\Driver\PDOSqlite\Driver',
+                        'params' => array(
+                            'memory' => true,
+                        ),
+                        'doctrineTypeMappings' => array(
+                            'money' => 'money',
```

Comment:
@tommyseus can you help me to understand this part of code? Without this map doesn't work